### PR TITLE
Ex1#2 add "move" and "destructor" special member methods 

### PR DESCRIPTION
**Pull request by idoa-DI**

### CODE REVIEW

File: headers/Ex1.hpp [hunk 1]
```diff
@@ -1,65 +1,72 @@
...
-    ~DynamicBuffer() noexcept;
+    DynamicBuffer<T>& operator=(DynamicBuffer<T>&& other) noexcept; 
+
+    ~DynamicBuffer() noexcept = default;
 
 private:
-    T* Allocate(size_t size);
-    void DeAllocate() noexcept;
-    void CopyDynamicBuffer(DynamicBuffer<T>& src);
+    void CopyDynamicBuffer(const DynamicBuffer<T>& src);
```
Comment (idoa-DI) [inline headers/Ex1.hpp]: methods' names should be in `camelCase`

File: headers/Ex1.hpp [hunk 2]
```diff
@@ -1,65 +1,72 @@
...
-
-    m_size = 0;
-    delete[] m_buffer;
-    m_buffer = nullptr;
-}
+/****************************************** Helper Functions *****************************************/
 
 template <typename T>
-void DynamicBuffer<T>::CopyDynamicBuffer(DynamicBuffer<T>& src) {
+void DynamicBuffer<T>::CopyDynamicBuffer(const DynamicBuffer<T>& src) {
```
Comment (idoa-DI) [inline headers/Ex1.hpp]: No need to do something about it, just think about it - Would this compile for T which has only move assignment?

File: headers/Ex1.hpp [hunk 3]
```diff
@@ -1,65 +1,72 @@
+#include <memory> //unique_ptr
+
 template <typename T>
 class DynamicBuffer {
```
Comment (idoa-DI) [inline headers/Ex1.hpp]: Please as you mentioned in the discussion of the previous exercise - add methods for accessing the objects in the buffer.

File: headers/Ex1.hpp [hunk 4]
```diff
@@ -1,65 +1,72 @@
...
 
 private:
-    T* Allocate(size_t size);
-    void DeAllocate() noexcept;
-    void CopyDynamicBuffer(DynamicBuffer<T>& src);
+    void CopyDynamicBuffer(const DynamicBuffer<T>& src);
 
     size_t m_size;
-    T* m_buffer;
+    std::unique_ptr<T[]> m_buffer;
```
Comment (idoa-DI) [inline headers/Ex1.hpp]: Don't use unique_ptr for  this exercise.

**Review by ItamarB-DI — APPROVED** [review]

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (idoa-DI) [inline .vscode/settings.json]: please add .vscode to your `.gitignore` file.
Those files has nothing to do with the PR

File: headers/Ex1.hpp [hunk 6]
```diff
@@ -1,68 +1,123 @@
+#include <memory> //unique_ptr
```
Comment (idoa-DI) [inline headers/Ex1.hpp]: there is no use for unique_ptr in this file. this include isn't needed

File: DynamicBuffer/Ex1.hpp [hunk 7]
```diff
@@ -0,0 +1,123 @@
+#include <memory> //unique_ptr
```
Comment (idoa-DI) [inline DynamicBuffer/Ex1.hpp:1]: there is no use for this header as you do not use unique_ptr

File: .vscode/settings.json (hunk 5 above)
Comment (idoa-DI) [reply .vscode/settings.json]: As you can see - this file is still included in branch.

**Review by ItamarB-DI — APPROVED** [review]

**Pull request merged**